### PR TITLE
Fix panic reading/writing `any` types

### PR DESCRIPTION
**Pull request by Jeffail**
Fixes a few edge cases where conversions to and/or from `any` types would panic due to boxed reflect.Value types. I've added a unit test to run through the examples I've come across and plugged all the panics from those, but I'm fairly confident there will be other (even more edgier) cases lurking in there.

### CODE REVIEW

**Comment by Jeffail** [comment]
I'm fairly sure this would partly fix some of the issues seen in https://github.com/parquet-go/parquet-go/issues/42

File: parquet_test.go [hunk 1]
```diff
@@ -1026,3 +1026,158 @@ var quickCheckConfig = quick.Config{
 func quickCheck(f interface{}) error {
 	return quickCheckConfig.Check(f)
 }
+
+func TestParquetAnyValueConversions(t *testing.T) {
+	// This test runs conversions to/from any values with edge case schemas.
+
+	type obj = map[string]any
+	type arr = []any
```
Comment (Jeffail) [inline parquet_test.go:1034]: I apologise in advance for these local aliases, definitely optimising here for writing at the cost of reading so happy to remove these if asked.

Comment (achille-roussel) [reply parquet_test.go:1034]: This doesn't bother me, thanks for calling it out 👍

File: row.go [hunk 2]
```diff
@@ -444,6 +444,10 @@ func deconstructFuncOfOptional(columnIndex int16, node Node) (int16, deconstruct
 func deconstructFuncOfRepeated(columnIndex int16, node Node) (int16, deconstructFunc) {
 	columnIndex, deconstruct := deconstructFuncOf(columnIndex, Required(node))
 	return columnIndex, func(columns [][]Value, levels levels, value reflect.Value) {
+		if value.Kind() == reflect.Interface {
+			value = value.Elem()
+		}
```
Comment (achille-roussel) [inline row.go:449]: Same here: should we `nil` check?

File: node.go [hunk 3]
```diff
@@ -292,6 +292,9 @@ type groupField struct {
 func (f *groupField) Name() string { return f.name }
 
 func (f *groupField) Value(base reflect.Value) reflect.Value {
+	if base.Kind() == reflect.Interface {
+		base = base.Elem()
+	}
```
[the remark this answers is not in the document]
Comment (Jeffail) [reply node.go]: It'd definitely be beneficial in terms of preventing panics. However, you could consider the provided data in that case to be invalid as the columns would not be marked as optional at that point. I've implemented the nil checks and added some more test cases to illustrate the behaviour. From my perspective this is good enough, but in the commit message I've noted that it might be considered more "correct" to reject the data with a polite error message, but I believe that'd require a refactor.

File: row.go (hunk 2 above)
Comment (Jeffail) [reply row.go:449]: @achille-roussel added but as mentioned above I think there's also a case for omitting it. If so just let me know and I'll purge that commit.

**Review by achille-roussel — APPROVED** [review]
Thanks for the fix and adding tests, this looks good to me 👍
